framework/src/mesh/MooseMesh.C

```diff
@@ -435,6 +439,101 @@ MooseMesh::update()
   _face_info_dirty = true;
 }
 
+void
+MooseMesh::buildLowerDMesh()
+{
+  auto & mesh = getMesh();
+
+  unsigned int max_n_sides = 0;
+
+  // remove existing lower-d element first
+  std::set<Elem *> deleteable_elems;
+  for (auto & elem : mesh.element_ptr_range())
+    if (elem->subdomain_id() == Moose::INTERNAL_SIDE_LOWERD_ID ||
+        elem->subdomain_id() == Moose::BOUNDARY_SIDE_LOWERD_ID)
+      deleteable_elems.insert(elem);
+    else if (elem->n_sides() > max_n_sides)
+      max_n_sides = elem->n_sides();
+
+  for (auto & elem : deleteable_elems)
+    mesh.delete_elem(elem);
+
+  deleteable_elems.clear();
+
+  dof_id_type max_elem_id = mesh.max_elem_id();
+  unique_id_type max_unique_id = mesh.parallel_max_unique_id();
+
+  std::vector<Elem *> side_elems;
+  _higher_d_elem_side_to_lower_d_elem.clear();
+  for (const auto & elem : mesh.active_element_ptr_range())
+  {
+    for (unsigned short int side = 0; side < elem->n_sides(); side++)
+    {
+      Elem * neig = elem->neighbor_ptr(side);
```

Comment:
You should probably have a check in here to skip this neighbor if it is remote_elem

Reply:
No, intentionally not skipping.

Reply:
Also boundary elements

Reply:
I don't think you follow. There's a difference between ghosted elements and remote elements. You can't store information on remote elements

Reply:
oh, right. Sorry, I was thinking ghosted elements.

Reply:
Reply @fdkong here: yes, I should not assume no other lower-d elements even though I deleted all lower-d elements with the two special subdomain ids.

Reply:
Great, thanks. I made another comment for the necessary addition to resolve this. Therefore, I'm resolving this convo